release.sh

```diff
@@ -28,7 +28,14 @@ if ! grep '^globus\-compute\-sdk \& globus\-compute\-endpoint v'"$VERSION"'$' do
 fi
 
 echo "releasing v$VERSION"
-git tag -s "$VERSION" -m "v$VERSION"
+if git tag -s -m "v$VERSION" "$VERSION" ; then
+  echo "Git tagged $VERSION"
+else
+  read -p "Tag $VERSION already exists.  Release packages with this tag anyway? [y/n] " -n 1 -r
+  if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+    [[ "$0" = "$BASH_SOURCE" ]] && exit 1 || return 1 # handle exits from shell or function but don't exit interactive shell
```

Comment:
FWIW, if using Bash's conditional (`[[`) as opposed to the `test` binary (`[`), then the quotes aren't necessary.  They don't hurt, but they aren't necessary:
```bash
    [[ $0 = $BASH_SOURCE ]] ...
```